src/py42/clients/archive.py

```diff
@@ -86,16 +93,102 @@ def stream_from_backup(
+            accepting_device_guid (str): The GUID of the device accepting the restore.
+            restore_path (str): The path on the accepting device where the restore will be saved.
+            destination_guid (str, optional): The GUID of the destination that stores the backup
+                of the file. If None, it will use the first destination GUID it finds for your
+                device. 'destination_guid' may be useful if the file is missing from one of your
+                destinations or if you want to optimize performance. Defaults to None.
+            archive_password (str or None, optional): The password for the archive, if password-
+                protected. This is only relevant to users with archive key password security. Defaults
+                to None.
+            encryption_key (str or None, optional): A custom encryption key for decrypting an archive's
+                file contents, necessary for restoring files. This is only relevant to users with custom
+                key archive security. Defaults to None.
+            file_size_calc_timeout (int, optional): Set to limit the amount of seconds spent calculating
+                file sizes when crafting the request. Set to 0 or None to ignore file sizes altogether.
+                Defaults to 10.
+            show_deleted (bool, optional): Set to True to include deleted files when restoring a directory.
+                Defaults to None.
+
+        Returns:
+            :class:`py42.response.Py42Response`.
+        """
+        explorer = self._archive_accessor_factory.create_archive_accessor(
+            device_guid,
+            ArchiveExplorer,
+            destination_guid=destination_guid,
+            private_password=archive_password,
+            encryption_key=encryption_key,
+        )
+        backup_set_id = self._select_backup_set_id(
+            device_guid, explorer.destination_guid
+        )
+        file_selections = explorer.create_file_selections(
+            backup_set_id, file_paths, file_size_calc_timeout
+        )
+        pusher = self._archive_accessor_factory.create_archive_content_pusher(
+            device_guid,
+            accepting_device_guid,
+            private_password=archive_password,
+            encryption_key=encryption_key,
+            destination_guid=explorer.destination_guid,
+        )
+        return pusher.stream_to_device(
+            restore_path,
+            accepting_device_guid,
+            file_selections,
+            backup_set_id,
+            show_deleted,
+        )
+
+    def _select_backup_set_id(self, device_guid, destination_guid):
+        backup_sets = self.get_backup_sets(device_guid, destination_guid)[u"backupSets"]
+        if not backup_sets:
+            return None
+        for backup_set in backup_sets:
+            backup_set_id = backup_set[u"backupSetId"]
+            # 1 is the default backup set ID and is preferred.
```

Comment:
@timabrmsn @alanag13 Looks like the UI prefers the backupset of ID 1 when it exists.